src/form.js

```diff
@@ -446,23 +389,12 @@ define([
             this.fireChange(mug);
         },
         /**
-         * Walk merged data and control trees calling valueFunc for each mug
-         *
-         * If common nodes in merged tree branches are not in the same order
-         * then the order of nodes in the control tree will be preserved, and
-         * all nodes in the data tree that are not also in the control tree will
-         * be visited last.
```

Comment:
I assume this behavior has moved into the parser? (haven't gotten there yet) I think it's important that we preserve this behavior and explicitly mention it in the place where it is implemented.